src/box/txn.c

```diff
@@ -1237,15 +1241,6 @@ txn_commit_impl(struct txn *txn, enum txn_commit_wait_mode wait_mode)
 	if (txn_has_flag(txn, TXN_WAIT_SYNC)) {
 		struct txn_limbo_entry *limbo_entry = txn->limbo_entry;
 		assert(limbo_entry->lsn > 0);
-		/*
```

Comment:
I don't like, that we change the behavior in this "refactoring" patch. We didn't introduce `wait_mode` yet.

Before this patch tx waited for `CONFIRM` to be written to disk, I suppose, the behavior should remain the same for `COMPLETE` mode for now, we should explicitly wait for `CONFIRM` to be written in `txn_commit_impl`. Now the behavior is incorrect, `req->is_complete` loop in `commit_impl` will exit, when we'll say a limbo worker to write a CONFIRM, but we won't wait for it to do so.

Reply:
I don't get your comment. We still have `txn_limbo_wait_complete`, which waits for the limbo entry confirmation.

Reply:
You're right, my bad. No problem here, we're blocked until `confirm` is written